### PR TITLE
Update booking type to match API response

### PR DESCRIPTION
**Pull request by pezholio**
This updates the Booking type to better match what comes back from the API. At the moment, dev is broken.

### CODE REVIEW

File: integration_tests/pages/manage/booking/show.ts [hunk 1]
```diff
@@ -16,8 +16,8 @@ export default class BookingShowPage extends Page {
 
   shouldShowBookingDetails(booking: Booking): void {
     cy.get('dl[data-cy-expected-dates]').within(() => {
```
Comment (richpjames) [inline integration_tests/pages/manage/booking/show.ts:18]: should we update this attribute too?

File: server/testutils/factories/booking.ts [hunk 2]
```diff
@@ -86,8 +87,8 @@ class BookingFactory extends Factory<Booking> {
 
 export default BookingFactory.define(() => ({
   person: personFactory.build(),
-  arrivalDate: faker.date.soon().toISOString(),
-  departureDate: faker.date.future().toISOString(),
+  arrivalDate: formatISO(faker.date.soon(), { representation: 'date' }),
+  departureDate: formatISO(faker.date.future(), { representation: 'date' }),
```
Comment (richpjames) [inline server/testutils/factories/booking.ts:91]: I'm happy to implement it but shouldn't we use the same date format across all our factories?

**Review by richpjames — APPROVED** [review]
Looks good, left a couple of comments

File: server/testutils/factories/booking.ts (hunk 2 above)
Comment (pezholio) [reply server/testutils/factories/booking.ts:91]: Yeah, I kinda want to pick these things up as we encounter them, but I think we'll need to.

File: integration_tests/pages/manage/booking/show.ts (hunk 1 above)
Comment (pezholio) [reply integration_tests/pages/manage/booking/show.ts:18]: I think it's fine, as far as the user is concerned they're expected dates. We just need to make the internal naming consistent

Comment (pezholio) [reply integration_tests/pages/manage/booking/show.ts:18]: Actually, as this data attribute is an internal thing, it probably does make sense to be consistent here too! 😆

**Pull request merged**